### PR TITLE
feat: add earn screen. add earn deeplinks

### PR DESCRIPTION
**Pull request by mitchellv-ledger**
### 📝 Description

Adds Earn dashboard screen and navigator.  Adds deeplinks specific for earn dashboard to open staking:
```
/**
* ie: "ledgerlive://earn" will open earn dashboard page
*
* @params ?action: string
* ie: "ledgerlive://earn?action=stake" will open staking flow
*
* * @params ?action: string
* * @params &accountId: string
* ie: "ledgerlive://earn?action=stake-account&accountId=XXXX" will open staking flow with specific account
*
*  * * @params ?action: string
* ie: "ledgerlive://earn?action=get-funds" will open buy drawer
*
* * @params ?action: string
* * @params ?currencyId: string
* ie: "ledgerlive://earn?action=get-funds&currencyId=ethereum" will open buy drawer with currency
*
*/
```

### ❓ Context

- **Impacted projects**: `LLM` 
- **Linked resource(s)**: `LIVE-7453` 

### ✅ Checklist

- [ ] **Test coverage** 
- [X] **Atomic delivery** 
- [X] **No breaking changes** 

### 📸 Demo

Can test deeplinks on iOS by using the command: `xcrun simctl openurl booted ledgerlive://earn?action=stake`
### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            if (account) {
+              navigation.navigate(NavigatorName.StakeFlow, {
+                screen: ScreenName.Stake,
+                params: {
+                  account,
+                  parentRoute: route,
+                },
+              });
+            } else {
+              console.log("not account found in earn dashboard deeplink");
```
Comment (sarneijim) [inline apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:65]: Not sure if we must track this kind of errors in Sentry...(not as a scope in this pr)

Comment (sarneijim) [inline apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:65]: Not sure if we must track these log errors in Sentry...(not as a scope in this pr)

**Comment by sarneijim** [comment]
Checking a little what we have in LLD I think we need to implement the logic to support "alwaysShowNoFunds" too, not sure if this is part of another ticket.

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import { EarnScreen } from "../../screens/PTX/Earn";
+import { BaseNavigatorStackParamList } from "./types/BaseNavigator";
+import { getAccountIdFromWalletAccountId } from "@ledgerhq/live-common/wallet-api/converters";
+import { accountsSelector } from "../../reducers/accounts";
+
+const Stack = createStackNavigator<EarnLiveAppNavigatorParamList>();
+
+const Earn = (_props: StackNavigatorProps<EarnLiveAppNavigatorParamList, ScreenName.Earn>) => {
+  // Earn dashboard feature flag
+  // TODO: update to use specific mobile feature flag
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:22]: I think we need to get this mobile ff in before we can merge. Is Xaviar on this?

Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:22]: might be good to add one on dev/staging first https://console.firebase.google.com/u/1/project/ledger-live-development/config

Comment (sarneijim) [reply apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:22]: Xavier can config ptxEarn to work different in android and ios platform. We do not need to create a new one.

Comment (beths-ledger) [reply apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:22]: ah good to know! Thanks Sara!

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx (hunk 1 above)
Comment (cng-ledger) [reply apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:69]: I think Sentry has a capture message feature, which can keep track of these error messages that isn't an exception

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            console.log("accountId query is missing for earn dashboard deeplink");
+          }
+          break;
+        }
+        case "get-funds": {
+          if (currencyId) {
+            navigation.navigate(NavigatorName.Exchange, {
+              screen: ScreenName.ExchangeBuy,
+              params: { currency: currencyId },
+            });
```
[the remark this answers is not in the document]
Comment (mitchellv-ledger) [reply apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx]: updated to navigate to no funds flow

**Review by sarneijim — APPROVED** [review]

**Pull request merged**